### ISSUE TITLE
Add support for incremental updating to Tap-GitLab

### ISSUE THREAD

**Issue by pnadolny13** [MeltanoLabs/tap-gitlab#9]
In GitLab by @zj-gitlab on Jun 14, 2019, 04:42

For now I've limited my data to the `gitlab-org` group, and a small project in that group. Now when I would add `gitlab-ce` to the projects I fear the updating of the data would take very long. Making this tap less interesting to use for large projects or groups.

/cc @iroussos

**Comment by pnadolny13** [comment]
In GitLab by @iroussos on Jun 20, 2019, 14:21

mentioned in merge request !14

**Comment by pnadolny13** [comment]
In GitLab by @iroussos on Jun 20, 2019, 14:46

closed via merge request !14

**Comment by pnadolny13** [comment]
In GitLab by @iroussos on Jun 20, 2019, 14:46

@zj all the issues you had should have been fixed with the latest updates. Check meltano/tap-gitlab!14 for more info.

The update speeds up both the initial execution of the Tap and all subsequent executions by properly using both the start date and the STATE of the previous run.

I just extracted one month of data from `gitlab-org/gitlab-ce` in ~30 minutes, including all the Epics for `gitlab-org`, the overhead from extracting the labels for the project and the Group and the fact that a lot of issues (4480), MRs (1763) and Epics (~500) are marked as updated during the last 20 days.
